tcms/core/tests/test_ajax.py

```diff
@@ -35,6 +44,48 @@ def test_empty_string_is_converted_to_bool(self):
         response = self.client.get(url)
         self.assertEqual(200, response.status_code)
 
+    def test_with_ulli_format(self):
+        """ When a request comes with info_type=categories, and format=ulli for given product_id,
+            we expect to receive all categories for that product, rendered as HTML unordered list
+        """
+
+        url = '%s?info_type=categories&product_id=%d&format=ulli' % (reverse('ajax-info'),
+                                                                     self.product.pk)
+
+        response = self.client.get(url)
+
+        self.assertContains(response, '<li>' + self.category_one.name + '</li>')
+        self.assertContains(response, '<li>' + self.category_two.name + '</li>')
+
+    def test_with_unrecognisable_infotype(self):
+        """ When a request comes with invalid info_type,
+            we expect to receive response containing the 'Unrecongnizable infotype' error message
+        """
+
+        url = "%s?info_type=INVALID" % reverse('ajax-info')
+
+        response = self.client.get(url)
+
+        self.assertEqual(str(response.content, encoding=settings.DEFAULT_CHARSET),
+                         'Unrecognizable infotype')
+
+    def test_with_json_format(self):
+        """ When a request comes with info_type=categories for given product_id,
+            we expect to receive all categories for that product as array of JSON objects """
+
+        url = "%s?info_type=categories&product_id=%d" % (reverse('ajax-info'), self.product.pk)
+
+        response = self.client.get(url)
+
+        self.assertIn(
+            json.loads('{"model": "testcases.category", "pk": %s, "fields": {"name": "%s"}}'
+                       % (self.category_one.pk, self.category_one.name)),
```

Comment:
This is a scary monster of a code for the expected value. It is a dict which is constructed inside of the test. There is no need to start from a string, replace string placeholders, validate through JSON to actually arrive at dict. Start with a clearly defined dict instead and assign it to `expected` or some such.